api/machine/machine.proto

```diff
@@ -713,3 +715,23 @@ message EtcdMemberList {
   repeated string members = 2;
 }
 message EtcdMemberListResponse { repeated EtcdMemberList messages = 1; }
+
+// rpc generateConfiguration
+// GenerateConfigurationRequest describes a request to generate a new configuration
+// on a node.
+message GenerateConfigurationRequest {
+  int32 machine_type = 1;
+  string version = 2;
+  string cluster_name = 3;
+  string endpoint = 4;
+  string kubernetes_version = 5;
+  string install_disk = 6;
+  string install_image = 7;
+  string dns_domain = 8;
+  string hostname = 9;
+}
+// GenerateConfiguration describes the response to a generate configuration request.
+message GenerateConfigurationResponse {
+  common.Metadata metadata = 1;
+  bytes data = 2;
```

Comment:
I think it might be nice to support here multiple files to be returned (with filenames). this would help with multi-doc config later on

Reply:
:ok: